src/cartesian_mesh.cc

```diff
@@ -1,31 +1,17 @@
 #include "flyft/cartesian_mesh.h"
+#include <iostream>
 
 namespace flyft
 {
 
-CartesianMesh::CartesianMesh(double L, int shape, double area)
-    : CartesianMesh(L,shape,area,0)
-    {
-    }
-
-CartesianMesh::CartesianMesh(double L,int shape, double area, double origin)
-    : Mesh(L,shape,origin), area_(area)
-    {
-    }
-
-CartesianMesh::CartesianMesh(int shape, double step, double area)
-    : CartesianMesh(shape,step,area,0)
-    {
-    }
-
-CartesianMesh::CartesianMesh(int shape, double step,double area, double origin)
-    : Mesh(shape,step,origin),area_(area)
+CartesianMesh::CartesianMesh(double lower,double upper,int shape, BoundaryType lower_bc, BoundaryType upper_bc,double area)
+    : Mesh(lower,upper,shape,lower_bc,upper_bc), area_(area)
     {
     }
 
 std::shared_ptr<Mesh> CartesianMesh::slice(int start, int end) const
     {
-    return std::shared_ptr<Mesh>(new CartesianMesh(end-start,step_,area_,lower_bound(start)));
+    return std::shared_ptr<Mesh>(new CartesianMesh(lower_bound(start),end,end-start,lower_bc_, upper_bc_,area_));
```

Comment:
This is not quite right. It also needs to account for the possible change in boundary conditions on slicing. I think this will work:
```suggestion
    return std::shared_ptr<Mesh>(new CartesianMesh(
        lower_bound(start),
        lower_bound(end),
        end-start,
        (start > 0) ? BoundaryType::internal : lower_bc_,
        (end < shape_-1) ? BoundaryType::internal : upper_bc_,
        area_
    ));
```